app/assets/javascripts/discourse/tests/unit/utils/decorators-test.js

```diff
@@ -175,4 +177,78 @@ module("Unit | Utils | decorators", function (hooks) {
 
     assert.strictEqual(stub.otherCounter, 1);
   });
+
+  test("@observes works via .extend and native class syntax", async function (assert) {
+    let NativeClassWithObserver;
+    withSilencedDeprecations("discourse.utils-decorators-observes", () => {
+      NativeClassWithObserver = class extends EmberObject {
+        counter = 0;
+        @observes("value")
+        incrementCounter() {
+          this.set("counter", this.counter + 1);
+        }
+      };
+    });
+
+    const ExtendsWithObserver = EmberObject.extend({
+      counter: 0,
+      @observes("value")
+      incrementCounter() {
+        this.set("counter", this.counter + 1);
+      },
+    });
+
+    const nativeClassTest = NativeClassWithObserver.create();
+    nativeClassTest.set("value", "one");
+    await settled();
+    nativeClassTest.set("value", "two");
+    await settled();
+    assert.strictEqual(
+      nativeClassTest.counter,
+      2,
+      "observer triggered for native class"
+    );
+
+    const extendsTest = ExtendsWithObserver.create();
+    extendsTest.set("value", "one");
+    await settled();
+    extendsTest.set("value", "two");
+    await settled();
+    assert.strictEqual(
+      extendsTest.counter,
+      2,
+      "observer triggered for extends"
+    );
+  });
+
+  test("@on works via .extend and native class syntax", async function (assert) {
+    let NativeClassWithOn;
+    withSilencedDeprecations("discourse.utils-decorators-on", () => {
+      NativeClassWithOn = class extends EmberObject {
+        counter = 0;
+        @on("init")
+        incrementCounter() {
+          this.set("counter", this.counter + 1);
+        }
+      };
+    });
+
+    const ExtendsWithOn = EmberObject.extend({
+      counter: 0,
+      @on("init")
+      incrementCounter() {
+        this.set("counter", this.counter + 1);
+      },
+    });
+
+    const nativeClassTest = NativeClassWithOn.create();
+    assert.strictEqual(
+      nativeClassTest.counter,
+      1,
+      "on triggered for native class"
+    );
+
+    const extendsTest = ExtendsWithOn.create();
+    assert.strictEqual(extendsTest.counter, 1, "on triggered for extends");
```

Comment:
```suggestion
    assert.strictEqual(extendsTest.counter, 1, "on triggered for extend");
```

(vars and class names can stay as is 😉)

Reply:
Ok, I fixed all of them now (including class/var names). Thanks!